utils/src/main/java/org/nervos/ckb/crypto/secp256k1/ECKeyPair.java

```diff
@@ -44,31 +44,54 @@ public static ECKeyPair createWithKeyPair(KeyPair keyPair) {
   }
 
   public static ECKeyPair createWithPrivateKey(BigInteger privateKey, boolean compressed) {
-    return new ECKeyPair(privateKey, Sign.publicKeyFromPrivate(privateKey, compressed));
+    return new ECKeyPair(privateKey, publicKeyFromPrivate(privateKey, compressed));
   }
 
   public static ECKeyPair createWithPrivateKey(BigInteger privateKey) {
-    return new ECKeyPair(privateKey, Sign.publicKeyFromPrivate(privateKey));
+    return new ECKeyPair(privateKey, publicKeyFromPrivate(privateKey));
   }
 
   public static ECKeyPair createWithPrivateKey(byte[] privateKey) {
     return createWithPrivateKey(Numeric.toBigInt(privateKey));
   }
 
   /**
-   * Sign a hash with the private key of this key pair.
+   * Returns public key from the given private key.
    *
-   * @param transactionHash the hash to sign
-   * @return An {@link ECDSASignature} of the hash
+   * @param privateKey the private key to derive the public key from
+   * @return BigInteger encoded public key
    */
-  public ECDSASignature sign(byte[] transactionHash) {
-    ECDSASigner signer = new ECDSASigner(new HMacDSAKCalculator(new SHA256Digest()));
+  public static BigInteger publicKeyFromPrivate(BigInteger privateKey) {
+    return publicKeyFromPrivate(privateKey, true);
+  }
 
-    ECPrivateKeyParameters privKey = new ECPrivateKeyParameters(privateKey, Sign.CURVE);
-    signer.init(true, privKey);
-    BigInteger[] components = signer.generateSignature(transactionHash);
+  public static BigInteger publicKeyFromPrivate(String privateKeyHex) {
+    return publicKeyFromPrivate(Numeric.toBigInt(privateKeyHex), true);
+  }
+
+  /**
+   * Returns public key from the given private key.
+   *
+   * @param privateKey the private key to derive the public key from
+   * @return BigInteger encoded public key
+   */
+  public static BigInteger publicKeyFromPrivate(BigInteger privateKey, boolean compressed) {
+    ECPoint point = publicPointFromPrivate(privateKey);
+
+    byte[] encoded = point.getEncoded(compressed);
+    if (compressed) {
+      return new BigInteger(1, Arrays.copyOfRange(encoded, 0, encoded.length));
```

Comment:
```suggestion
      int from = compressed ? 0 : 1;
      return new BigInteger(1, Arrays.copyOfRange(encoded, from, encoded.length));
```
Seems those two line only has a diff in from, will this be more clear?

Reply:
Fixed https://github.com/nervosnetwork/ckb-sdk-java/pull/149/commits/bf1b0d4dd6c02e973bf5285afaeb402105593971